src/EditorFeatures/CSharpTest/ExtractClass/ExtractClassTests.cs

```diff
@@ -0,0 +1,1443 @@
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">
+class Test
+{
+    int [||]MyProperty { get; set; }
+}
+        </Document>
+    </Project>
+</Workspace>";
+
+            var expected = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">
+class Test : MyBase
+{
+}
+        </Document>
+        <Document FilePath=""MyBase.cs"">internal class MyBase
+{
+    int MyProperty { get; set; }
+}</Document>
+    </Project>
+</Workspace>";
+
+            await TestAsync(input, expected);
+        }
+
+        [Fact]
+        public async Task TestField()
+        {
+            var input = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">
+class Test
+{
+    int [||]MyField;
+}
+        </Document>
+    </Project>
+</Workspace>";
+
+            var expected = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">
+class Test : MyBase
+{
+}
+        </Document>
+        <Document FilePath=""MyBase.cs"">internal class MyBase
+{
+    int MyField;
+}</Document>
+    </Project>
+</Workspace>";
+
+            await TestAsync(input, expected);
+        }
+
+        [Fact]
+        public async Task TestFileHeader()
+        {
+            var input = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">// this is my document header
+// that should be copied over
+
+class Test
+{
+    int [||]Method()
+    {
+        return 1 + 1;
+    }
+}
+        </Document>
+    </Project>
+</Workspace>";
+
+            var expected = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">// this is my document header
+// that should be copied over
+
+class Test : MyBase
+{
+}
+        </Document>
+        <Document FilePath=""MyBase.cs"">// this is my document header
+// that should be copied over
+
+internal class MyBase
+{
+    int Method()
+    {
+        return 1 + 1;
+    }
+}</Document>
+    </Project>
+</Workspace>";
+
+            await TestAsync(input, expected);
+        }
+
+        [Fact]
+        public async Task TestWithInterface()
+        {
+            var input = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">
+interface ITest 
+{
+    int Method();
+}
+
+class Test : ITest
+{
+    int [||]Method()
+    {
+        return 1 + 1;
+    }
+}
+        </Document>
+    </Project>
+</Workspace>";
+
+            var expected = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">
+interface ITest 
+{
+    int Method();
+}
+
+class Test : MyBase, ITest
+{
+}
+        </Document>
+        <Document FilePath=""MyBase.cs"">internal class MyBase
+{
+    int Method()
+    {
+        return 1 + 1;
+    }
+}</Document>
+    </Project>
+</Workspace>";
+
+            await TestAsync(input, expected);
+        }
+
+        [ConditionalFact(AlwaysSkip = "https://github.com/dotnet/roslyn/issues/45977")]
```

Comment:
wjhy skip?  can we run the test (with the broken behavior) but have issue tracking fixing the behavior?

Reply:
With broken behavior is just verify it doesn't trigger the refactoring. I'd prefer to show what the desired behavior is as a test to unblock for #45977